python/ray/tune/ray_trial_executor.py

```diff
@@ -565,45 +576,26 @@ def restore(self, trial, checkpoint=None):
         """
         if checkpoint is None or checkpoint.value is None:
             checkpoint = trial.checkpoint
-        if checkpoint is None or checkpoint.value is None:
-            return True
+        if checkpoint.value is None:
+            return
         if trial.runner is None:
-            logger.error(
-                "Trial %s: Unable to restore - no runner. "
-                "Setting status to ERROR.", trial)
-            self.set_status(trial, Trial.ERROR)
-            return False
-        try:
-            value = checkpoint.value
-            if checkpoint.storage == Checkpoint.MEMORY:
-                assert type(value) != Checkpoint, type(value)
-                trial.runner.restore_from_object.remote(value)
-            else:
-                logger.info("Trial %s: Attempting restoration from %s", trial,
-                            checkpoint.value)
-                with warn_if_slow("get_current_ip"):
-                    worker_ip = ray.get(trial.runner.current_ip.remote(),
-                                        DEFAULT_GET_TIMEOUT)
-                with warn_if_slow("sync_to_new_location"):
-                    trial.sync_logger_to_new_location(worker_ip)
-                with warn_if_slow("restore_from_disk"):
-                    ray.get(
-                        trial.runner.restore.remote(value),
-                        DEFAULT_GET_TIMEOUT)
-        except RayTimeoutError:
-            logger.exception(
-                "Trial %s: Unable to restore - runner task timed "
-                "out. Setting status to ERROR", trial)
-            self.set_status(trial, Trial.ERROR)
-            return False
-        except Exception:
-            logger.exception(
-                "Trial %s: Unable to restore. Setting status to ERROR", trial)
-            self.set_status(trial, Trial.ERROR)
-            return False
-
+            raise RuntimeError(
+                "Trial {}: Unable to restore - no runner found.".format(trial))
+        value = checkpoint.value
+        if checkpoint.storage == Checkpoint.MEMORY:
+            assert type(value) != Checkpoint, type(value)
+            trial.runner.restore_from_object.remote(value)
+        else:
+            logger.info("Trial %s: Attempting restore from %s", trial, value)
+            with warn_if_slow("get_current_ip"):
+                worker_ip = ray.get(trial.runner.current_ip.remote(),
+                                    DEFAULT_GET_TIMEOUT)
+            with warn_if_slow("sync_to_new_location"):
+                trial.sync_logger_to_new_location(worker_ip)
+            with warn_if_slow("restore_from_disk"):
+                ray.get(
```

Comment:
Hm, I think this `ray.get` was actually only a stop-gap. We shouldn't actually need to wait for the entire setup in the event loop (we don't wait in `restore_from_object` anyways). This does not affect the rest of your PR though; just decreases the chance we hit the timeout.

Reply:
I'm also not sure if `restore` is necessarily idempotent, which this PR assumes. By removing the timeout, we can avoid this assumption.

Reply:
hmm i figured idempotency was a reasonable assumption to make but regardless yeah it would be good to get rid of the blocking restore here. i think the issue is that we need to know whether or not the restore actually worked and if not, restart from scratch—so if we don't `get` here we need to somewhere higher up in the stack to make sure the restore worked.